desc/equilibrium/coords.py

```diff
@@ -453,43 +448,57 @@ def _partial_sum(lmbda, L_lmn, omega, W_lmn, iota):
     return lmbda_minus_iota_omega, jnp.fft.rfftfreq(grid.num_theta, 1 / grid.num_theta)
 
 
-@partial(jit, static_argnames=["tol", "maxiter"])
-def _map_clebsch_coordinates(
+@partial(jit, static_argnames=["inbasis", "outbasis", "tol", "maxiter"])
+def _map_poloidal_coordinates(
     iota,
-    alpha,
+    poloidal,
     zeta,
     L_lmn,
     lmbda,
+    varepsilon=None,
+    inbasis=("rho", "alpha", "zeta"),
+    outbasis=("rho", "theta", "zeta"),
     guess=None,
     *,
     tol=1e-6,
     maxiter=30,
     **kwargs,
 ):
-    """Find θ for given Clebsch field line poloidal label α.
+    """Map poloidal coordinate in the input basis to the output basis.
+
+    If ``varepsilon`` is not given, then the input coordinate grid is constructed as
+    the tensor product of the values ``iota``, ``poloidal``, and ``zeta``.
 
-    # TODO: input (rho, alpha, zeta) coordinates may be an arbitrary point cloud
-    #       and the partial summation will work without modification.
-    #       Clean up input parameter API to support this.
+    If ``varepsilon`` is given, then the input coordinate grid is taken to be the
+    meshgrid of values ε = α + ι ζ stored in that array.
 
     Parameters
     ----------
-    iota : ndarray
-        Shape (num iota, ).
-        Rotational transform.
-    alpha : ndarray
-        Shape (num alpha, ).
-        Field line labels.
-    zeta : ndarray
+    iota : jnp.ndarray
+        Shape (num ρ, ).
+        Rotational transform ι(ρ) on flux surfaces labeled by ρ.
+    poloidal : jnp.ndarray
+        Shape (num poloidal, ).
+        Values for coordinate specified by ``inbasis[1]``.
+    zeta : jnp.ndarray
         Shape (num zeta, ).
-        DESC toroidal angle.
+        DESC toroidal angle ζ.
     L_lmn : jnp.ndarray
         Spectral coefficients for λ.
     lmbda : Transform
-        Transform for λ built on DESC coordinates [ρ, θ, ζ].
+        Transform for λ built on DESC coordinates (ρ, θ, ζ) uniformly spaced in θ.
+    varepsilon : jnp.ndarray
+        Shape should broadcast with shape (num ρ, num poloidal, num ζ).
+        Optional meshgrid of values ε = α + ι ζ.
+        This meshgrid need not be a tensor-product grid.
```

Comment:
what exactly does this mean? in most places we've been using meshgrid to mean a tensor product grid

Reply:
Meshgrids are more general than tensor-product grids. For example, if you iterate along the the second axis of the quantity ``varepsilon.reshape(num rho, num poloidal, num zeta)``, then the values of ``varepsilon[:, i, :]`` can be different for each ``i``. In contrast, on a tensor-product grid, varepsilon has to be the same for each `i`.

The purpose of the comment is basically to inform the user that we support tensor product grids in (rho, alpha, iota zeta) in addition to tensor product grids in (rho, alpha, zeta). 

If you want to perform a coordinate mapping for a single poloidal transit on every flux surface, then giving ``varepsilon`` as a meshgrid in (rho, alpha, zeta) works but it can't be a tensor-product grid in (rho, alpha, zeta) unless iota is constant wrt to rho.